src/core/net/nat64_translator.cpp

```diff
@@ -656,6 +656,7 @@ void Translator::UpdateState(bool aAlwaysNotify)
 
     if (aAlwaysNotify)
     {
+        mState = newState;
```

Comment:
This fix is correct and fine.

However, after reading the related code, I have a suggestion to simplify this.

It appears we want to signal a change whenever the CIDR is updated, regardless of any state change and the `aAlwaysNotify` is used to implement this.

We can simplify this by modifying `Translator::SetIp4Cidr()` to directly signal:

* **From:**
    ```c++
    // Always notify the platform when the CIDR is changed.
    UpdateState(true /* aAlwaysNotify */);
    ```

* **To:**
    ```c++
    UpdateState();

    // Notify the platform when the CIDR is changed.
    Get<Notifier>().Signal(kEventNat64TranslatorStateChanged);
    ```

Calling `Signal` multiple times won't cause issues, as the `Notifier` accumulates events before reporting them via the "state changed callback" (from its own `Tasklet`).

We can implement this change in a separate PR. 

Thoughts? @superwhd @erjiaqing

Reply:
I agree with this suggestion. It looks easier to understand. 👍 

I think the change should be simple enough so maybe I can update this PR to apply the suggestion.

Reply:
Updated. Please take another look. Thanks!

Reply:
Thanks @superwhd for changing it in this PR. Looks great. 👍